src/connectComp.cpp

```diff
@@ -0,0 +1,265 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#include <vector>
+#include <iostream>
+
+#include <unordered_map>
+
+#include <unordered_set>
+#include <string.h>
+#include <cassert>
+#include <cstdlib>
+
+namespace found {
+
+struct Point {
+	int x;
+	int y;
+};
+
+struct Edge {
+	int magnitude; // number of points in this edge
+	std::vector<Point> points;
+	// int Xmin, Xmax, Ymin, Ymax;
+};
+
+std::vector<Edge> NaiveConnectedComponent(unsigned char *image, int imageWidth, int imageHeight) { // const??
+	int threshold = 3; // over threshold means possible components exist
+
+	// Tells us if certian numbers in the below array are part of the same component
+    std::unordered_map<int, int> equivalencies;
+
+	// make an array with the same dimensions of the image, but it shows connected components using numbers
+    unsigned char *edges = (unsigned char *) std::malloc(imageWidth * imageHeight * sizeof(unsigned char));
+    edges[0] = (image[0] > threshold) ? 1 : 0;
+    std::cout << static_cast<int>(edges[0]) << std::endl;
```

Comment:
I bet this print statement shouldn't be here. Eventually, when you're done debugging and testing, I will ask that you remove all print statements that shouldn't exist in the program, as well as "commented-out" code.